### ISSUE TITLE
Add `pigz`

### ISSUE THREAD

**Issue by jeffwidman** [actions/runner-images#8205]
### Tool name

pigz

### Tool license

MIT

### Add or update?

- [X] Add
- [ ] Update

### Desired version

Any version is fine

### Approximate size

259KB

### Brief description of tool

`pigz` is parallel gzip:
* https://zlib.net/pigz/
* https://github.com/madler/pigz

It used to be included by default in the runner images, but it turns out that was an accidental side effect:
* https://github.com/actions/runner-images/issues/8161#issuecomment-1693247211

We (the :dependabot: team) use this for an internal deploy workflow. We can of course manually install `pigz` in our workflow, and it's simple to do so, but I was surprised to notice that takes 2 mins just to install. That nearly doubles the length of our deploy pipeline! So we'd rather have it installed by default and save the time.

You can see from the issue that even though it was filed only a few days ago, it already has multiple 👍 indicating that others relied on `pigz` as well.

### URL for tool's homepage

https://zlib.net/pigz/

### Platforms where you need the tool

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
- name: Install pigz
  run: sudo apt-get install -y pigz
```

### Tool installation time in runtime

2 minutes

### Are you willing to submit a PR?

Yes!

**Comment by vpolikarpov-akvelon** [comment]
Hey @jeffwidman. We will consider installing this package, but according to my tests it takes only 15 seconds to install it in workflow.

**Comment by vpolikarpov-akvelon** [comment]
According to the log the most of the time is taken by man-db trigger processing. I found information about a bug in man-db ([1858777](https://bugs.launchpad.net/ubuntu/+source/man-db/+bug/1858777)) that was fixed in Ubuntu 22.04, but wasn't in Ubuntu 20.04. I think it may be the cause.

The simplest workaround would be just removing man-db from the system: `sudo apt-get remove -y --purge man-db`.

Long story short, I don't think there is something wrong with your setup, you just were out of luck :)

---

And yeah, we added `pigz` to runner image; it will be available when the next release rollout is complete

**Comment by mikhailkoliada** [comment]
Added to the image :)

**Comment by jeffwidman** [comment]
FWIW, we (the :Dependabot: team) have now switched to `zstd` as we found it provided better compression than `gzip` w/o losing the parallelism of `pigz`:
* https://github.com/dependabot/dependabot-core/pull/7950

So we no longer need `pigz` installed in the runner image.
